modules/core/src/main/java/org/apache/synapse/message/store/impl/rabbitmq/RabbitMQProducer.java

```diff
@@ -36,147 +36,186 @@
-			isConnectionError = true;
-		} catch (Throwable t) {
-			throwable = t;
-			error = true;
-		} finally {
-			if (channel != null && channel.isOpen())
-				try {
-					channel.close();
-				} catch (TimeoutException e) {
-					logger.error("Error when closing connection,TimeoutException" + synCtx.getMessageID() + ". " , e);
-				} catch (IOException e) {
-					logger.error(
-							"Error when closing connection" + synCtx.getMessageID() + ". " + e);
-				}
-		}
-		if (error) {
-			String errorMsg = getId() + ". Ignored MessageID : " + synCtx.getMessageID()
-			                  + ". Could not store message to store ["
-			                  + store.getName() + "]. Error:" + throwable.getLocalizedMessage();
-			logger.error(errorMsg, throwable);
-			store.closeProducerConnection();
-			connection = null;
-			if (logger.isDebugEnabled()) {
-				logger.debug(getId() + ". Ignored MessageID : " + synCtx.getMessageID());
-			}
-			return false;
-		}
-		if (logger.isDebugEnabled()) {
-			logger.debug(getId() + ". Stored MessageID : " + synCtx.getMessageID());
-		}
-		store.enqueued();
-		return true;
-	}
-
-	/**
-	 * Used to close the channel opened in this object instance.
-	 * This should be called after the end of each call on storeMessage method
-	 * But instead of this, finally block is used in storeMethod to close the channels
-	 */
-	public boolean cleanup() {
-		return store.cleanup(null, false);
-	}
-
-	public boolean isInitialized() {
-		return isInitialized;
-	}
-
-	public String getId() {
-		return idString;
-	}
+    private static final Log log = LogFactory.getLog(RabbitMQProducer.class.getName());
+
+    private static final int DEFAULT_PRIORITY = 0;
+    private Connection connection;
+    private RabbitMQStore store;
+    private String routingKey;
+    private String exchangeName;
+    private boolean isInitialized = false;
+    private String idString; // ID of the MessageProducer
+
+    /**
+     * The RabbitMQ producer
+     *
+     * @param store the {@link RabbitMQStore} object
+     */
+    public RabbitMQProducer(RabbitMQStore store) {
+        if (store == null) {
+            log.error("Cannot initialize.");
+            return;
+        }
+        this.store = store;
+        isInitialized = true;
+    }
+
+    /**
+     * Store the given message into the queue and return whether the operation success or not
+     *
+     * @param synCtx Message to be saved.
+     * @return {@code true} if storing of the message is successful, {@code false} otherwise.
+     */
+    @Override
+    public boolean storeMessage(MessageContext synCtx) {
+        if (synCtx == null) {
+            return false;
+        }
+        if (connection == null) {
+            log.error(getId() + " cannot proceed. RabbitMQ Connection is null. Ignored MessageId: " +
+                    synCtx.getMessageID());
+            return false;
+        }
+        boolean result = false;
+        try (Channel channel = connection.createChannel()) {
```

Comment:
I belive `connection.createChannel()` is not a heavy operation? Otherwise, this will effect the performance. We need to keep the channel in a variable and reuse.

Reply:
I see an exception unhandled on my IDE

![image](https://user-images.githubusercontent.com/4967046/82746902-6310b400-9db2-11ea-8e14-70ab08282f24.png)

Reply:
As per the store mediator design, a channel has to create and close per message.

`produceStatus = messageStore.getProducer().storeMessage(newCtx);`

The above error might be due to rabbitmq client library not updated properly.